### PR TITLE
Adds coupons to cargo, turns small crates into goodies

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
Adds coupons to the back of cigarette packets! Once you empty a packet of cigs, use it in hand to rip off a coupon, good for anywhere from 10-50% off one goody crate! What's a goody crate, you ask?

Small item crates are now goodies! Things like single eguns and single medkits are now in their own menu called goodies, and can only be purchased with private accounts. They come in lockboxes locked to the purchaser's account rather than crates, so they don't have to have that stupid 700 point crate tax put on them which made it impossible for average crewmembers to afford stuff. Mind you this is only for goody items, normal crates still have crates.

Also adds a few new goody crates for people to buy, including valentines cards and bee plushies! There's lots more opportunities!

## Why It's Good For The Game
Lets people actually afford small items from cargo, fixes single medkit crates letting you make infinite money since they're worth less than the crate they're shipped in, encourages smoking

## Changelog
:cl: Ryll/Shaps
add: Cigarette packets now have coupons on the back for small cargo items! Smoking DOES pay!
tweak: Single/small items in cargo have been rebranded as goodies, come in lockboxes rather than crates, and can only be purchased with private accounts. They also are immune to the 10% private account fee, obviously.
/:cl:

### CODE REVIEW

**Comment by Shadowflame909** [comment]
Finally, items are now affordable to the crew.

**Comment by Cartographer-D** [comment]
I don't like the idea that you can purchase weapons as goodies, but otherwise great.

**Comment by ATH1909** [comment]
this buffs the smoker quirk smh my head my head

**Comment by Ryll-Ryll** [comment]
Weapons were already available as goodies, if anything this makes them slightly less available since you can't use the cargo budget to buy the single crates anymore

**Comment by zxaber** [comment]
are multiple single items purchased by one person still grouped into the same container?

**Comment by Ryll-Ryll** [comment]
Yep, so theoretically you could stuff a bunch of combat shotguns in one carryable case if you buy like 6 at once, but you can't reinsert them once you've taken them out, and single shotguns are still 3k credits each soo

**Comment by ATH1909** [comment]
> Yep, so theoretically you could stuff a bunch of combat shotguns in one carryable case if you buy like 6 at once, but you can't reinsert them once you've taken them out, and single shotguns are still 3k credits each soo

inB4 someone buys like 20 syndicate briefcases full of cash, then just starts Reaper-ing shotguns from a case that they drag behind them

**Comment by Archanial** [comment]
What if you got them for completing bounties that are related to your job?

**Comment by Ryll-Ryll** [comment]
That would be beyond the scope of this PR, as far as I'm aware cargo actually sends off all the bounties, so they'd have to manually distribute the coupon to the person responsible. Maybe there could be a way to scan your ID when you drop off a bounty at cargo and get PDA'd an e-coupon when the bounty actually ships, but for now that sounds like a separate PR in the future

**Comment by ArcaneMusic** [comment]
> 
> 
> What if you got them for completing bounties that are related to your job?



> 
> 
> That would be beyond the scope of this PR, as far as I'm aware cargo actually sends off all the bounties, so they'd have to manually distribute the coupon to the person responsible. Maybe there could be a way to scan your ID when you drop off a bounty at cargo and get PDA'd an e-coupon when the bounty actually ships, but for now that sounds like a separate PR in the future

In the economy hackMD I did mention I was looking into a system for crew bounties, I could reasonably see using this in order to support such a feature, but you're right that it's out of scope for this.

On the body of the PR, big ups. That said, I wish they weren't called goodies since that sounds like a 49 cent candybar that you get at Halloween, look at, thank the person, then never eat them and eventually throw them out.

File: code/modules/cargo/goodies.dm [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	cost = 150
+	contains = list(/obj/item/valentine, /obj/item/reagent_containers/food/snacks/candyheart)
+
+/datum/supply_pack/goody/beeplush
+	name = "Bee Plushie"
+	desc = "The most important thing you could possibly spend your hard-earned money on."
+	cost = 1500
+	contains = list(/obj/item/toy/plush/beeplushie)
+
+/datum/supply_pack/goody/camera
```
Comment (Tlaltecuhtli) [inline code/modules/cargo/goodies.dm:121]: cameras can be printed in autolathe, also no glubs crate?

**Comment by Tlaltecuhtli** [comment]
also coupons could be rewarded by joining as high priority job

**Comment by HomocidlePancakes** [comment]
So you know how junk food companies often have these giveaways, stuff like “Use the code inside the bag for a chance to win a PS4/vacation/car/1 million dollars!”. We can have the same. Here are some rewards I thought up:

-**1 million credits**
Small print on the back explains that you will be given 1 million over the next 5 years, so instead it just increases your wage by 500 bucks. Write your name on the back of the coupon and send it on the shuttle to redeem it. Of course, QM won’t erase your name and write his own instead when you hand it over.

-**Waffletendo 5DS**
A pocket arcade machine both games on it. You gain thrice the experience from it.

-**A golden plumber**
It’s a golden plumber. That’s it. Does 15 damage.

-**A hoverboard signed by Tony Eagle**
Tony Eagle is the number one skater in the 26th century. This board has been tuned by tony itself, so you can jump a whole one more tile further than a regular hoverboard

-**A vacation to w̴̛̼̰̺͚̣̩̤̞͈̬̲͈̉́͑̇̇̋̕͜ͅh̶͖͖̱̰̹̹̰̻̻͎̯́̃̈̍͊̒͘̚é̴̢͓͍̱̮̞̤͚̩͗͋͘̕͜n̵̢͎̱̼͚͚͕̯̖̪̗̙̬͚̂̅͘͠t̸̛͖͕̦̲͓͆͋͝ͅḩ̸̡̯̩͚̜̝͍̘͙̩̄e̵̹͉̭͚̜̫̺̩͛̔̌̏̈́̌̋͑̅̿̓m̸̨̡̧̘̞͕̯̣̫̂̉̃́̈̄̇͌̏̉̕ͅo̷̧͔̰̟̺͇͙̩̻̮̱̮͋͛͛̔̈͐̈́̍̋́̋̊̿̚͠ọ̷̼̥͈̖̱̬̳̙̲̻̙̤̏̿͜ͅn̸̨̖͇̪̜̮̄̾̄͝͝i̶̠͎͖̳̓͆̿͊̈́͂̑̐̉̔̀͑͝͝s̵̨̡̢̩̙̗̮͕̘̈́͜͝ͅi̴̢̢͈͚͕̯̮̲͙̬͕̇̈́̾̾̈́͑̆̍̑̚n̵̨̨̧̘̦̗̱̪͉̮͓͓̳̥̮͗̾̎ť̶̢̩̱̪̲̠̘̙̪̈́̅̄̕͘ͅh̵̢̢͔̻̖͎̣̥͈̣̓̈́̓ę̴̢̺͔̮͚̝͍̘͕̠̠̩̺͔̈́̽̀̿̇͑͆̒̋͛̕̕e̴̤͙̯̤̜̬͍̝̋̌̈́̒̆̓ȳ̴̥̠̘̥̰e̸̡̢̢̨̬̗͔̟͍̐͌̾̓̌͘͜͜ó̷̯̗̗̠͈f̵̮̻̫͍͇̜̲̀̏͆͋̊̊̂̃t̴̡̨͚̰̗̝͇̣̼̲͍̀̓́h̸̗́̽̓̽͊͑͘͝ē̴͙͈͆̐́́̓̀̇d̸̬̲̅͑́͑̆̍͌̂̾̂ȧ̶̯̙̰̫̬͔̻̠̬̣͑͋̚r̵̨̧͍̮͕̲̖͎̟̮̦͉̣͌͑̂͋k̵̡̛̼͓͍̖͖̖̬̥̞̟̲͚͒̏̑̉̅̂͑̽̓̒̓̚͝h̵̰̭̤̮͂̈́̽̾͌͗̈́͝a̷̢͓̳͔̦̘͉͙̰͛̈́͊͐̐̌̀́̒̊͊ẅ̷̛̳̲̱͓͙̠̗̦̳̯̪͕̪͎̞́̀̃̍̑̍̊k̵̗̱̀̉̿̑͝ț̸̹͚͖̓͒̎h̵͔̬͍͕̙̖͍̖̟̝̩̦̺̘͛̅͛̒ê̷̢͈̥̫͉̦̪͙̪̲̱̩̖̄͑̉̉͗͒̂͜ͅȯ̵̡̡̱̠̰̗̫͙̬̥̟͍̝̈́́̈́̍̄̾̏̀́̏ͅl̵̛̺̙̳̦͋̒̈́d̵̡̛̛͚̭̝̰̤̳͓̬̗̫̾̐̏̃̍̑͘͠o̴̹̹̭̘̮̗̠̻̩̥̰̜̫͍̊̋͐̐̐̌̇̃͊̐͘͝ͅn̵̖͕̫̞͈̗̙͍͋̒͋̓͒̃͗̐͗̚͝ḛ̴̰̈́̓̅͝ș̴̡̢͎͉̰̱̗͊́͗͜͝h̴̫̣̦̝͂̈́̎̈́̇̌̅̑̈́̕͝͠à̴̡̨̡͖̩͕̱̲͎̖̩͝ͅl̴͚̠̞͈͖̮͎͍̻͕̉͗̀͛͑̍̔͗̂͘͠͝l̶̨̘̻̳̰̣̜̙̰̦̖̼̻̆̉̀̃̏̚ŗ̷̧̣͖̦͎͎̪̠̼̫͎̼̹͐̉̿̆̂̈́̾̋́i̴̹̳̦̊̒͊̉̇̌͑̑͌́̈́̀̆͒͝ͅs̶̛͖̣̹̜̜͈̲̫͓̺̣͖̟̅͆̊̔̒͌̈́͆̅̿͌͝͝ͅe̷̹͌ḧ̸̨̫̲̭͓̲́̎̚e̵͍͑͂́̐͒͛̄̍̅̀͌̏́̇̋͜s̶̢̺͎̥̲͈̫̲̘͖̕h̴̨̙̜̖̀͘à̸̧̨̡̖̦̗͍̳̥͚̞̱̃͘͘l̵̬̐͊̾̀̽̋͒̀̎̃̓̿̆͂ļ̶̥̥̺̳͕̫͓̘̏͆̔̓̅c̸̡̡̙͕̣̥̠͍̞̦͇̪͙͚̃́͆͝͝ô̴̫̩̺̖͇̗͕̼̮̈́͗̈͒̆ṃ̸̛̥̇͐̋͂̿̒̕̚͘͝ë̸̯͔͎̥́́́͛́̀͒͗̇h̸̨̡̻̱̮̬̟͇̹̗̭̙̱̐̆̅̈́͆͗͋͜ͅẹ̴̡̢̡̯̥̘͕͈̭̞̺͖̣̇̔̔̏̔̊̎͊̊̈́̚͘͠s̸̳̤͔͔̠̩͔̦̲̃h̶̨̭͍̖͚͚͓͚͚͆̀̅̊̅̎͌̄̉̐͒ͅä̸̢̧͖͇̥̭̫́͋̊͗̅͂ļ̶̧̦̝̰̜̠̜̖̺̘̮̑̐͗̕ͅͅl̶̲̣̼͖̫͔̜̰̅̍͑̉̍̆̐̾͑̿́́͝c̷̡̘̬͔̗̮̖͎̦̪̼̃̆ó̴̹̿m̸͍̞̤͚͈̺͎̩̀̃͑̎̈́̒̓̊͆́͝͝e̸͕̺̗̣̭̾̈́́͜ě̵̡̲͙̦̺̘͚̈͊͆̈́ͅm̵̡̢̧̟̜̫͗̑̓̽̌̓̋b̶̨̡̳͖̫̖̺͖͇͚̩̱̝̈́̆̃͂̍̾̑̓̉͋̈́̚̚͘͜r̶͔̠̣̳̼̰̺̙̥̤͍͔̥̩̋̽̚ͅâ̴̘̂͋̊̉s̵̖͕̹͛̇͌̿̆̃̈͆̀͜͜͠ë̶̢͚̥͉́̈̋̋̏͊̈̓̅̒͑͜ͅh̷̨̨̰̱̰̥̼̜̦̺̀i̸̛͈͓̜̲͂́̅͌̋̏̊̿͂͝ḿ̵̨̡̯̘̘̖͕̖̟́̎̄̓́̋̾̃̓̊̈́̕͜͝**



*A vacation for you and your whole family! Disregard the bloodstains on the coupon. Just rip this in two and our representatives will instantly transport you and your loved ones to our 666 star beach resort.*

Does nothing for chaplains. When used in hand has a chance for one of these things to happen.

-You loose your soul, preventing revival.
-You get the “hand” brain trauma. You know, the one that chases you around the station.
-You explode in a shower of gore
-You cough up a random organ
-You get a dark blessing
-You get a heart of darkness
-You start to “see the truth” and get X-ray, along schizophrenia trauma.

File: code/datums/components/omen.dm [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		var/turf/the_turf = t
+		for(var/obj/machinery/vending/darth_vendor in the_turf)
+			if(darth_vendor.tiltable)
+				darth_vendor.tilt(our_guy)
+				qdel(src)
+				return
+
+/// If we get knocked down, see if we have a really bad slip and bash our head hard
+/datum/component/omen/proc/check_slip(mob/living/our_guy, amount)
+	if(amount <= 0 || !prob(50))
```
Comment (nemvar) [inline code/datums/components/omen.dm:55]: ``!prob(50)`` :smirk:

File: code/modules/cargo/goodies.dm (hunk 1 above)
Comment (Ryll-Ryll) [reply code/modules/cargo/goodies.dm:121]: oh fair then, and there's still the solo gamer mitts pack

File: code/datums/components/omen.dm (hunk 2 above)
Comment (Ryll-Ryll) [reply code/datums/components/omen.dm:55]: i swear when i first wrote this it wasn't a 50, it made sense then

File: code/game/objects/items/storage/fancy.dm [hunk 3]
```diff
@@ -138,6 +138,26 @@
...
+		attached_coupon = new
+
+/obj/item/storage/fancy/cigarettes/attack_self(mob/user)
+	if(contents.len == 0 && attached_coupon)
+		to_chat(user, "<span class='notice'>You rip the back off \the [src] and get a coupon!</span>")
+		user.put_in_hands(attached_coupon)
+		attached_coupon.generate()
+		attached_coupon = null
+		name = "discarded cigarette packet"
+		desc = "An old cigarette packet with the back torn off, worth less than nothing now."
```
[the remark this answers is not in the document]
Comment (nemvar) [reply code/game/objects/items/storage/fancy.dm:151]: This has not been resolved.
Add a ``spawn_coupon = FALSE`` here.

File: code/modules/shuttle/supply.dm [hunk 4]
```diff
@@ -123,12 +127,15 @@ GLOBAL_LIST_INIT(blacklisted_cargo_types, typecacheof(list(
 		value += SO.pack.cost
 		SSshuttle.shoppinglist -= SO
 		SSshuttle.orderhistory += SO
+		qdel(SO.applied_coupon)
```
Comment (nemvar) [inline code/modules/shuttle/supply.dm:130]:
```suggestion
		QDEL_NULL(SO.applied_coupon)
```

File: code/game/objects/items/storage/fancy.dm (hunk 3 above)
[the remark this answers is not in the document]
Comment (Ryll-Ryll) [reply code/game/objects/items/storage/fancy.dm:151]: ack, sorry!!

**Pull request merged**